### ISSUE TITLE
Deploy changed application builds for staging in master CI

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va.gov-team#29142]
## Description

For staging in master CI we are running a full build and deploy. We need to change this to run the same partial build we're using in the PR CI and sync the artifacts to S3 instead of doing a normal full deployment.

We still need to ability to run a full staging deployment for global upgrades or fixing any issues. This can be a manual option passed into the GitHub action to run and deploy the full build to S3.

## Definition of Done

- [x] master CI run partial staging build
- [x] master CI deploys only the built/changed files to S3 for staging
- [x] Full staging build/deploys are still possible

**Comment by theodur** [comment]
Full staging builds & deploys are possible with the [manual-deploy-dev-staging](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/manual-deploy-dev-staging.yml) workflow

**Comment by theodur** [comment]
Changed app builds and deploys have been enabled for staging in `master`. Going through the [test-plan](https://docs.google.com/spreadsheets/d/1Lz3cfJhs_oorT8mk5yEMwcpEZWPToFsCQZgChGH31DE/edit#gid=0) for staging

**Comment by theodur** [comment]
I noticed an issue where deployed apps weren't able to fetch chunk ids:

![image (2).png](https://images.zenhubusercontent.com/5e56a58c13bedb092ce69379/e13eeb6a-3699-4b5c-a865-8acf30601c9e)

This is due to changed app builds generating different chunk ids than full builds. Since we're only syncing the changed app bundles and named Webpack chunks, but not the chunk ids, the assets from changed app builds expect the chunk ids that were created in it's build to be available, when they aren't.

When the Webpack mode is set to `development`, chunk ids are automatically named:

![image.png](https://images.zenhubusercontent.com/5e56a58c13bedb092ce69379/2fa31518-c708-4fae-869d-736299958c50)

But when the Webpack is set to `production`, chunk ids are not named:

![image.png](https://images.zenhubusercontent.com/5e56a58c13bedb092ce69379/abcbf408-fac3-41a3-ac7d-f4dd7666a085)

This is why we are not seeing the same issue on dev, because the chunk ids are named, so the changed app builds and full builds are referencing the same chunk ids, rather than ones with dynamic ids that can vary.

The chunk ids can be set to `named` for production mode in [Webpack](https://webpack.js.org/configuration/optimization/#optimizationchunkids) which would enable chunk ids to be named like in dev.

Also, as seen in the first screenshot, some apps like VAOS have their own chunk ids, so the `remove-global-assets` script will be updated to sync those too.

**Comment by theodur** [comment]
I opened [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/20759) for updating the Webpack chunk ids to be named based off the file path instead of random ids. This should allow the chunk ids between full builds and single app builds to be consistent

**Comment by theodur** [comment]
It's also worth pointing out that if/when we turn `vets-website` into a monorepo, or enable teams to use their own dependencies, it might make sense for each app to have its own folder in the `generated` directory, where the app will use assets from. This would help with the isolation of dependencies, and also alleviate the need to remove global assets, since we could just sync the changed app build assets to it's corresponding directory in `generated`.

**Comment by theodur** [comment]
After running into an issue with changed app deployments, where the deployed app was referencing a non-existent property, I discovered it was due to the `moduleIds` changing between full and single app builds. Because the `moduleId` isn't consistent across builds, single app builds generate different ids.

I merged a [PR](https://github.com/department-of-veterans-affairs/vets-website/pull/20855) for setting the `moduleIds` option to `named` in the Webpack config. This should allow the `moduleIds` to be consistent for single app builds
